File: platform/polyglot/gemini_openai_sdk/py/gemini_openai_sdk/health_check.py
"""
Health Check Module

Performs health check by making a minimal LLM call using the flash model.
"""

import time
from typing import Any, Dict, Optional

from .client import chat_completion
from .constants import DEFAULT_MODEL, MODELS, SYSTEM_PROMPT
from .logger import create

logger = create("gemini_openai_sdk", __file__)


async def health_check(
    model_type: Optional[str] = None,
    system_prompt: Optional[str] = None,
) -> Dict[str, Any]:
    """
    Perform health check with LLM status verification.

    Args:
        model_type: Default model type
        system_prompt: System prompt

    Returns:
        Health status dictionary with LLM connectivity check
    """
    start_time = time.time()
    logger.debug("health_check: starting")

    model_type = model_type or DEFAULT_MODEL
    system_prompt = system_prompt or SYSTEM_PROMPT

    llm_status = "unknown"
    llm_error = None
    llm_response_time = None

    try:
        # Make a minimal LLM call using flash model for speed
        response = await chat_completion(
            messages=[{"role": "user", "content": "ping"}],
            model=MODELS["flash"],
            max_tokens=5,
            temperature=0,
        )

        llm_response_time = int((time.time() - start_time) * 1000)
        content = response.get("choices", [{}])[0].get("message", {}).get("content")
        llm_status = "connected" if content else "error"
        logger.debug("health_check: LLM ping successful", response_time_ms=llm_response_time)
    except Exception as err:
        llm_response_time = int((time.time() - start_time) * 1000)
        llm_status = "error"
        llm_error = str(err)
        logger.error("health_check: LLM ping failed", error=llm_error)

    total_time = int((time.time() - start_time) * 1000)

    return {
        "status": "healthy" if llm_status == "connected" else "unhealthy",
        "llm_status": llm_status,
        "llm_error": llm_error,
        "llm_response_time_ms": llm_response_time,
        "models": MODELS,
        "default_model": model_type,
        "system_prompt": system_prompt,
        "total_time_ms": total_time,
    }

File: platform/polyglot/gemini_openai_sdk/mjs/gemini-openai-sdk/health-check.mjs
/**
 * Health Check Module
 *
 * Performs health check by making a minimal LLM call using the flash model.
 */

import { create } from './logger.mjs';
import { MODELS, DEFAULT_MODEL, SYSTEM_PROMPT } from './constants.mjs';
import { chatCompletion } from './client.mjs';

const logger = create('gemini-openai-sdk', import.meta.url);

/**
 * Perform health check with LLM status verification.
 *
 * @param {object} options - Health check options
 * @param {string} [options.modelType] - Default model type
 * @param {string} [options.systemPrompt] - System prompt
 * @returns {Promise<object>} Health status with LLM connectivity check
 */
export async function healthCheck(options = {}) {
  const startTime = Date.now();
  logger.debug('healthCheck: starting');

  const modelType = options.modelType || DEFAULT_MODEL;
  const systemPrompt = options.systemPrompt || SYSTEM_PROMPT;

  let llmStatus = 'unknown';
  let llmError = null;
  let llmResponseTime = null;

  try {
    // Make a minimal LLM call using flash model for speed
    const response = await chatCompletion(
      [{ role: 'user', content: 'ping' }],
      {
        model: MODELS.flash,
        max_tokens: 5,
        temperature: 0,
      }
    );

    llmResponseTime = Date.now() - startTime;
    llmStatus = response.choices?.[0]?.message?.content ? 'connected' : 'error';
    logger.debug('healthCheck: LLM ping successful', { responseTimeMs: llmResponseTime });
  } catch (err) {
    llmResponseTime = Date.now() - startTime;
    llmStatus = 'error';
    llmError = err.message;
    logger.error('healthCheck: LLM ping failed', { error: err.message });
  }

  const totalTime = Date.now() - startTime;

  return {
    status: llmStatus === 'connected' ? 'healthy' : 'unhealthy',
    llm_status: llmStatus,
    llm_error: llmError,
    llm_response_time_ms: llmResponseTime,
    models: MODELS,
    default_model: modelType,
    system_prompt: systemPrompt,
    total_time_ms: totalTime,
  };
}

export default healthCheck;
